src/ngraph/op/result.cpp

```diff
@@ -44,15 +42,10 @@ void op::Result::validate_and_infer_types()
 
 shared_ptr<Node> op::Result::copy_with_new_args(const NodeVector& new_args) const
 {
-    if (new_args.size() != 1)
-    {
-        throw ngraph_error("Incorrect number of new arguments");
-    }
+    check_new_args_count(this, new_args, 1);
 
-    if (new_args.at(0)->get_outputs().size() != 1)
-    {
-        throw ngraph_error("Result::copy_with_new_args expected a single-output argument");
-    }
+    NODE_VALIDATION_ASSERT(this, new_args[0]->get_outputs().size() == 1)
```

Comment:
I have preserved this check, but I think it is redundant. (The condition will be checked at validation.)

Reply:
We should drop it -- in the future if the input node hasn't been validated yet, the test will fail.